plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/ui/navigator/model/AppEngineDispatchDescriptor.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.facets.ui.navigator.model;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.wst.common.project.facet.core.IFacetedProject;
+
+/**
+ * Represents a {@code dispatch.xml} element.
+ */
+public class AppEngineDispatchDescriptor extends AppEngineResourceElement {
```

Comment:
Why are these classes so small? There were a lot fo long if blocks above to select between them. Just a thought, but could any of that logic be pushed into these classes instead?

Reply:
+1